.github/workflows/generate-index-and-docs.yml

```diff
@@ -0,0 +1,41 @@
+name: Check and generate PURL Type Docs and Index
+
+on:
+  push:
+    paths:
+      - "types/*.json"
+      - "schemas/*.json"
+      - "etc/"
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  generate-index-and-docs:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install dependencies
+        run: make conf
+
+      - name: Validate code and data formats
+        run: make check
+
+      - name: Generate index and docs
+        run: make gendocs
+
+      - name: Commit and push changes
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git add types/ types-doc/
+          git commit -s -m "Generate updated PURL type documentation" || echo "No changes to commit"
+          git push
```

Comment:
Isn't `main` a protected branch? Direct commits to protected branches will fail.

A better approach would be to follow a pattern used by other GitHub Actions-based projects (for example [`check-uncommitted.yml`](https://github.com/dependabot/fetch-metadata/blob/main/.github/workflows/check-uncommitted.yml) in `dependabot/fetch-metadata`):

1. **Shift the responsibility to the developer.**
   Require users to run the documentation generation script before submitting a PR.

2. **Update the CI workflow to only check for cleanliness:**
   Modify this workflow to regenerate the documentation **but fail if that introduces changes** (i.e., if the working tree becomes dirty).

Reply:
@ppkarwasz Let's handle this separately. @stevespringett contributed that initially but I am not sure I want the commits to happen automatically just yet...